### PR TITLE
refactor(http): use Http.request for all http shorthand methods

### LINKED ISSUE #11670: Http shorthand methods should call this.request
**I'm submitting a ...**  (check one with "x")
```
[ ] bug report => search github for a similar issue or PR before submitting
[x] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

Since HTTP interceptors are no longer a thing in Angular 2, to get something similar, it is best to override the `Http` service to introduce something like an `APIService` (which is nicer anyway). Now let's say the APIService should always set an API token to be send as an `Authorization` header. This could be as simple as as only overriding the `request` method, if all other shorthand methods simply delegated to `request()` with `return this.request()`. But instead, they call the internal function `httpRequest`, which cannot be overridden. This means a lot of boilerplate, because we have to override every single shorthand method. It would be a non-breaking change to change this.

You can see it in this example (first google result for "angular2 http interceptor"): https://www.illucit.com/blog/2016/03/angular2-http-authentication-interceptor/

That example also uses a `getRequestOptionArgs` method to normalize the options before modifying (because you can pass a hash, `Request` object etc). It would be nice to have this as a method aswell, so we don't have to reimplement it.

Source: https://github.com/angular/angular/blob/master/modules/%40angular/http/src/http.ts#L136-L138

### PR DESCRIPTION
**Pull request by DzmitryShylovich**
Closes https://github.com/angular/angular/issues/11670

### CODE REVIEW

**Comment by vicb** [comment]
Thanks, could you add a test ?

**Comment by DzmitryShylovich** [comment]
What kind of test a can add? 

- There is test verifying that `request` method [accepts a fully-qualified request as its only parameter](https://github.com/angular/angular/blob/master/modules/%40angular/http/test/http_spec.ts#L112).
- There are tests for [all shorthand methods](https://github.com/angular/angular/blob/master/modules/%40angular/http/test/http_spec.ts#L202-L324).
I haven't introduced any new behavior.

**Comment by vicb** [comment]
if there is no new behavior then this PR is useless ?

**Comment by DzmitryShylovich** [comment]
@vicb updated tests

File: modules/@angular/http/test/http_spec.ts [hunk 1]
```diff
@@ -217,6 +220,7 @@ export function main() {
            inject([AsyncTestCompleter], (async: AsyncTestCompleter) => {
              backend.connections.subscribe((c: MockConnection) => {
                expect(c.request.method).toBe(RequestMethod.Post);
+               expect(http.request).toHaveBeenCalled();
```
Comment (vicb) [inline modules/@angular/http/test/http_spec.ts]: please make sure it has not been called before initiating the request

**Comment by vicb** [comment]
:+1:
LGTM with one small update required.

Thanks for all your contribs lately !

**Comment by DzmitryShylovich** [comment]
done

**Pull request merged**